Adding commonly used prime editors and base editors.

ethanj801 opened the pull request:
Added a suite of commonly used prime and base editors that are optimized for human cells (although they are functional in bacteria).

jakebeal commented:
Can you please remove the .DS_Store files? They are Apple hidden files that shouldn't be added to the repository.

Also are you able to enable the actions to run on your repository? That will add some commits that turn the Excel file into reviewable CSV diffs, as well as checking the formatting and producing the distribution exports.

ethanj801 commented:
@jakebeal Would you mind taking a look at the structure of the excel file/.gb files and letting me know what I am doing wrong. I assume I am specifying something incorrectly, which is causing failure at the vector build stage. 

If a summary helps, I've added 6 additional rows (each containing a CDS part) to the table. I added information to each column and an associated .gb file for each row (each .gb is matched to the Data Source ID column). I've also added each CDS to the existing row in libraries and composites tab.

jakebeal commented:
It looks like there's an issue with the naming.  My guess would be `ABE8.20-m`, since GenBank uses "." to mean versioning, so the conversion might not be working correctly. If that isn't it, then it might be `ABE8e(TadA-8e V106W)` since parentheses are also unusual in a UID.

The system should be able to give us better information about such a problem but, well, it isn't: that's an infrastructure issue to open.

ethanj801 commented:
Ok. Is this something that should be fixed on my end in terms of changing those names? I do know the "." notation is very common with some groups who evolve enzymes.

jakebeal commented:
I think let's go both routes at once. Please adjust the names on your end to see if the error goes away. If it does go away, then please open an issue that indicates that names of this type are causing bugs.
If it doesn't go away, then the bug may be stickier...

ethanj801 commented:
Ok I've tried both and neither resolves the issue. How are sequences supposed to be specified for a given part? I was assuming it was .gb (unless you use a the registry as a prefix in which case it grabs from there), but I notice that for example in the 2A peptides there aren't any sequences in the rows or any references to the .gb files or data prefixes. Not sure if I am misunderstanding something fundamental about how the excel sheets work.

jakebeal commented:
It looks like the problem now is that you updated the parts sheet but not the composites sheet. It's failing earlier with a clear error message about the problem:

> Could not export SBOL file for package CRISPR-Cas: Could not resolve subparts
 in 'Cas variants': 'ABE8e(TadA-8e V106W)' 'SaABE8e(TadA-8e V106W)'

ethanj801 commented:
Ok. So the issue was in fact with the "(" not with the ".". This is actually preferable because for these particular parts it was not too big of a deal to remove the "(".

jakebeal commented:
Excellent!  From an informatics perspective, I can approve this pull request now. I'll leave it to @vinoo-igem or others to review the parts from a biological perspective (you might ask @GC-repeat , who put the first portion of the package in)

Can you please also open an issue for dealing with names with parentheses in them? We should either allow them or provide an error that explains what the problem is. I suspect this might apply to other non-standard characters too.

ethanj801 commented:
Issue opened.
#151

GC-repeat commented:
@ethanj801 for the prime editing system is there a canonical gRNA scaffold sequence (without the specific parts) that we could add in the package description ?

ethanj801 commented:
@GC-repeat Yes. There is a scaffold (which is functionally the same as the traditional gRNA scaffold with a 3' extension for the insertion). This addgene (https://www.addgene.org/132777/) plasmid is from the original publication and is designed for golden gate insertion of three oligos components (Spacer/Target, Scaffold and 3' extension [Primer Binding Site and RT template]). The complete protocol can be found in Supplementary note 3 in the original article.  https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-019-1711-4/MediaObjects/41586_2019_1711_MOESM1_ESM.pdf

The canonical sequence of a pegRNA consists of a spacer for targeting (for SpCas 20bp, typically starting with a G in the U6 addgene system above for transcription purposes). Followed by the scaffold (from supplementary table 3):
GTTTTAGAGCTAGAAATAGCAAGTTAAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGC 

And then followed by the 3' extension, which consists of a primer binding site (it is recommended to start with 13nts for this, but efficiency varies by site and is best tested empirically, though I know there is ongoing work on calculators and model predictions) and an RT template (typically 20nt, but at least 10 usually, again done empirically). Importantly, the 1st base of the 3' extension should not be a C, as this causes majorly decreased editing efficiency due to base stacking. This is usually not an issue since you can just slide your edit slightly up or downstream easily due to the flexibility of the system. In the cases where it is an issue, my understanding is that an alternative scaffold sequence can be used where the first and last bases of the terminal hairpin are inverted. That sequence is:
GTTTTAGAGCTAGAAATAGCAAGTTAAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGGACCGAGTCGGTCC

When using PE3b, the nicking guides are just standard guide RNAs (you use the same enzyme as the prime editor has nickase ability). 

Let me know if there are any questions.

jakebeal commented:
@ethanj801 Can you please add the critical information to the package description so that a package user doesn't need to find and read this thread?

GC-repeat commented:
@ethanj801
This gives something like this ?
5'-gRNA_spacer- GTTTTAGAGCTAGAAATAGCAAGTTAAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGC-RT_template-Primer_Binding_Site-3'
For "the 1st base of the 3' extension should not be a C", is it between the scaffold and the RT_template or the last 3' base ?

ethanj801 commented:
> @ethanj801 This gives something like this ? 5'-gRNA_spacer- GTTTTAGAGCTAGAAATAGCAAGTTAAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGC-RT_template-Primer_Binding_Site-3' 

Yes I believe this to be correct, though I'd appreciate if someone else would take a look at the original source documents and make sure I'm correct. The Liu Lab has a [guide on designing pegRNAs](https://drive.google.com/file/d/1xHWhscwIsyxUUdyGO-BZ8kleYUVmpBJA/view) that might also be useful for reference. As is often mentioned in these papers, designing a pegRNA is somewhat complex due to having ensure the orientation of all the sequences are correct. Typically what I've done (and what I know Liu Lab did ~1.5 years ago) is use an excel sheet or other automated tool to simplify manual design. Though due to the high variability in efficiencies between different pegRNAs and the number of the different designs that could be created, it might be worth recommending that naive users just use an automated tool such as [PrimeDesign](https://drugthatgene.pinellolab.partners.org/).

> For "the 1st base of the 3' extension should not be a C", is it between the scaffold and the RT_template or the last 3' base ?

From the main text of the original paper (emphasis mine)
> a C as the first nucleotide of the 3′ extension can disrupt guide RNA structure by pairing with G81.... Because many RT template lengths support prime editing, we recommend designing pegRNAs so that **the first base of the 3′ extension is not C**

GC-repeat commented:
Ok, I understand it the same way so with RT_template : 5'-DNNNNNNNNNNNNNNNNNNNN-3' (D = A,G,T, **not** C)

ethanj801 commented:
> @ethanj801 Can you please add the critical information to the package description so that a package user doesn't need to find and read this thread?

Done.

jakebeal commented:
Thanks, @ethanj801 . I am good with this being merged as soon as @GC-repeat approves.

GC-repeat commented:
Looks good to me, just two small questions : 
- with the removal of the parenthesis that were an issue I think the names are less clear : ABE8e(TadA-8e V106W) -> ABE8e TadA-8e V106W. Would it be better to use hyphen/underscore to see more easily what goes with what, ABE8e-TadA-8e_V106W or ABE8e_TadA-8e-V108W for example ?
- Are all the editing windows relative to a PAM at location 21-23 ? Is SaABE8e... PAM also at position 21?

ethanj801 commented:
> * Are all the editing windows relative to a PAM at location 21-23 ? Is SaABE8e... PAM also at position 21?
All of the SpCas9 editors certainly are. I adopted that language since it was used in one of the original publications, but perhaps it isn't worth the trouble? The 1 position is the same as first base of the spacer in all cases. For SaABE8e I believe the PAM would be position 23-28 (as I believe typically 22nt spacers are used).

ethanj801 commented:
Just corrected the issue with the `Final Product` column. Had to commit twice since a weird error showed up the first time i ran through the workflow. (Presumably something got weird with the excel sheet).

vinoo-igem commented:
@ethanj801 happy to approve and merge this! But honestly it would be on the basis of @GC-repeat as I'm unfamiliar with the vast majority of these new additions. Also wanted to flag that these larger enzymes will likely run into TWIST's synthesis upper limit, but having them as part of a package is useful going forward!
